hubbubhealth/idp-saml-example
# Update required rails gems with the vulnerability patch

This PR updates rails gems from `5.2.2` to `5.2.2.1` due to security updates based on GitHub warning alert related to the vulnerability that makes `Possible Remote Code Execution Exploit `.

# F
-[ ]  Update rails gems to `5.2.2.1`

# T


Use cases that were tested are:
- **Sign up without a hubbub account**
- **Sign up and create a hubbub account**
- **Sign up with an already existing hubbub account**
- **Log in to hubbub via IDP**
- **Change company via IDP app**


### Test case 1
### **Sign up without a hubbub account**

**Test Steps**
- Go to IDP app (https://hubbub-qa2-saml-idp.herokuapp.com/) and click "Sign up"
- Fill all the data and click "Sign up"

**Expected Result**
- User should be able to sign up via IDP app

**Actual Result**
- [ ] As expected 

**Test Data**
e-mail - `testidp@sink.sendgrid.net`
Password - `Password1@`

****
### Test case 2
### **Sign up and create a hubbub account**

**Test Steps**
- Go to IDP app (https://hubbub-qa2-saml-idp.herokuapp.com/) and click "Sign up"
- Fill all the data and click "Sign up"
- Click the "Saml initiate" link
- You should land in `https://qa.hubbubhealth.com/users/sso`
- Click "Create a new account"
- Enter a screen name, accept TOS and confirm email
- Click "Continue"

**Expected Result**
- User should be able to sign up via idp app
- User should be able to create a hubbub account via idp

**Actual Result**
- [ ] As expected 

**Test Data**
e-mail - `testidp01@sink.sendgrid.net`
Password - `Password1@`

****
### Test case 3
### **Sign up with an already existing hubbub account**

**Pre-Requirements**
- Having an existing and working hubbub account

**Test Steps**
- Go to IDP app (https://hubbub-qa2-saml-idp.herokuapp.com/) and click "Sign up"
- Fill all the data and click "Sign up"
- Click the "Saml initiate" link
- You should land in `https://qa.hubbubhealth.com/users/sso`
- Select "Yes", enter your hubbub email and pasword and click `Log in`
- Click `Connect`

**Expected Result**
- User should be able to sign up via idp app
- User should be able to sign up via idp and login to their existing hubbub account

**Actual Result**
- [ ] As expected 

**Test Data**
e-mail - `testidp02@sink.sendgrid.net`
Password - `Password1@`

****
### Test case 4
### **Log in to hubbub via idp**

**Pre-Requirements**
- Have a hubbub account and have signed up to idp app

**Test Steps**
- Go to IDP app (https://hubbub-qa2-saml-idp.herokuapp.com/) and click "Log in"
- Enter your email and password and click `Log in`
- Click the "Saml initiate" link

**Expected Result**
- User should be able to Login to hubbub via idp app

**Actual Result**
- [ ] As expected 

**Test Data**
e-mail - `testidp02@sink.sendgrid.net`
Password - `Password1@`

****
### Test case 5
### **Change company via idp app**

**Pre-Requirements**
- Have a hubbub account linked to a sponsor

**Test Steps**
- Go to IDP app (https://hubbub-qa2-saml-idp.herokuapp.com/) and click "Sign up"
- Fill all the data and click "Sign up" (note: company identifier should be `CAMBIA`)
- Click the "Saml initiate" link
- You should land in `https://qa.hubbubhealth.com/users/sso`
- Select "Yes", enter your hubbub email and password and click `Log in`
- Verify display message is correct `Your account was already connected to a company. Let's update that to your new company`
- Click `Connect`

**Expected Result**
- User is able to log in with the new company
- User should be able to change from one company to "Cambia Health Solutions" company

**Actual Result**
- [ ] As expected 

**Test Data**
e-mail - `testidp03@sink.sendgrid.net`
Password - `Password1@`
previous company `osiris`


#D
No actions are required
